llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Passes/TileAndDecomposeAttention.cpp

```diff
@@ -463,24 +464,14 @@ static void decomposeTiledAttention(IREE::LinalgExt::AttentionOp tiledAttnOp,
 
 /// Utility function which tiles and then decomposes attention op via
 /// FlashAttention algorithm.
-SmallVector<Operation *>
-tileAndDecomposeAttention(IREE::LinalgExt::AttentionOp attnOp,
-                          RewriterBase &rewriter, bool onlyTile) {
-  SmallVector<Operation *> ops = tileAttention(attnOp, rewriter);
+void tileAndDecomposeAttention(IREE::LinalgExt::AttentionOp attnOp,
+                               SmallVectorImpl<Operation *> &ops,
+                               RewriterBase &rewriter, bool onlyTile) {
+  IREE::LinalgExt::AttentionOp tiledAttentionOp =
+      tileAttention(attnOp, ops, rewriter);
   if (onlyTile)
-    return ops;
-  auto tiledAttnOp = cast<IREE::LinalgExt::AttentionOp>(ops[ops.size() - 1]);
-  ops.pop_back();
-  Operation *truncateToF16 = NULL;
-  Type elementType = tiledAttnOp.getQueryType().getElementType();
-  if (elementType.isF16()) {
-    truncateToF16 = ops[ops.size() - 1];
```

Comment:
I might not be tracking things properly, but what happened to this F16 truncation logic after the split here? Is it not needed anymore and/or was rolled into decomposition?